server/controllers/auth/login.ts

```diff
@@ -0,0 +1,36 @@
+import { Request, Response, NextFunction } from 'express';
+import { compare } from 'bcryptjs';
+
+import { checkUserLogin } from '../../queries/auth/checkUserLogin';
+import { validateLogin } from '../../utils/validation/auth/loginValidation';
+import { CustomError } from '../../utils/custom_error';
+import { signToken } from '../../utils/jwt';
+
+export const loginController = async (req: Request, res: Response, next: NextFunction): Promise<any> => {
+  try {
+    await validateLogin(req.body);
```

Comment:
you can use the output of the validate to get a typed output for the body!

Reply:
`const body = await validateLogin(req.body);`

Reply:
Yes, I can, but the idea is that since no error occurred, the body of the request is valid to use.

Reply:
Yes but if you use stuff like
trim() or uppercase()
you need to use the return